# Histone Supply: Multitiered Regulation Ensures Chromatin Dynamics Throughout the Cell Cycle

Citation: Mendiratta, S., Gatto, A., & Almouzni, G. (2019). Histone supply: Multitiered regulation ensures chromatin dynamics throughout the cell cycle. The Journal of cell biology, 218(1), 39–54. https://doi.org/10.1083/jcb.201807179

# Introduction

- **All histone families exist as variants that are differentially expressed and can undergo several post-translation modifications.**
- **A variety of nucleosomes thus use distinct histone variants, as well as post-translational modifications with different properties often associated with specific chromatin states.**
- **Nonnucleosomal histones are undercheck by their chaperones.**
- Expression peaks:
    - H3.1 & H3.2: During S phase
    - H3.3: During quiescent and terminally differentiated cells.
    - CenH3 CENP-A: G2/M phase in mammalian cells.

# Dedicated Chaperones for Dynamics and Deposition of Replicative and Non-replicative Histone Variants

- H3.3 is mainly associated with transcribed regions and regulatory sites with high nucleosome turnover but also, more broadly, at any given location where a gap occurs.

# Replicative Histone Variants: A Unique Genomic Organization and Regulation at Multiple Levels

- **Replicative histone variants share a unique transcriptional program that ensures higher expression levels for DNA synthesis-coupled deposition in the S phase**.
- Replicative histone variants genes show a peculiar organization in clusters that comprise multiple copies of all core histones and the H1 linker. This offers a potential means to optimize co-regulation.
- In the human and mouse genome, replicative histone genes cluster at three syntenic loci that remained physically linked through evolution. The human histone cluster 1 (HIST1) is located on chromosome 6 (6p22) and comprises more than 50 coding genes, while HIST2 and HIST3 are located on chromosome 1 (1q21 & 1q42) and contain respectively, 10 and 3 genes.
- The genes are also compartmentalized and processed in nuclear bodies that concentrate the factors required for their transcription and processing.
- **Replicative histones differ from replacement variants in terms of gene architecture and mRNA processing.**
- **Virtually all genes in histones clusters lack introns, have relatively short UTRs, and produce transcripts that harbor a conserved 3' stem-loop structure and do not undergo polyadenylation in many organisms.**
- Replicative histone genes are transcribed and processed by several factors within the histone locus bodies (HLBs).
- Higher-order structures can be revealed from self-interacting regions where contacts among genome sequences in physical proximity occur more frequently (topologically associating domains [TADs]).
- The three gene subsets in the human cluster 1 interact within separate TADs. Notably, these TADs further interact with each other over an ∼1.5-Mb distance, indicating that separate genes also establish long-range contacts and come in physical proximity within the nucleus
- In organisms with replicative histone variants, their expression throughout the cell cycle is controlled at multiple levels
- Transcription initiation is controlled by cyclin E/CDK2-dependent phosphorylation of the nuclear protein, ataxiatelangiectasia locus (NPAT) at the G1/S transition.
- Phosphorylated NPAT is required to initiate the assembly of HLBs and persists throughout S phase to activate the expression of replicative histone genes
- NPAT also interacts with FLICE-associated huge protein (FLASH), an essential cofactor involved in HLB assembly and 3′ processing of nascent transcripts
- Pre-mRNA maturation relies on the recognition of the cis-regulatory elements at the 3′ end by two factors that are specific to replicative histones: the stem-loop binding protein (SLBP) and the U7 snRNP
- The U7 snRNP is a ribonucleoprotein complex composed of U7 snRNA, Sm proteins, and U7-specific Lsm10 and Lsm11 proteins
- The 5′ end of the complex binds histone pre-mRNAs via U7 snRNA hybridization to the 3′ histone downstream element sequence
- SLBP binds the stem-loop upstream and interacts with the U7 snRNP to stabilize its association. This stabilization is required for proper maturation and cleavage of histone pre-mRNAs
- Pre-mRNA cleavage is catalyzed by the CPSF73 endonuclease
- The interaction between FLASH and the U7-specific protein Lsm11 is critical for CPSF73 recruitment at the cleavage site
- SLBP, U7 snRNP, and FLASH contribute to recruit scaffolding factors and additional components of the histone cleavage complex. The core components include Symplekin and CPSF100 which heterodimerizes with CPSF73 to catalyze the endonucleotyc cleavage of the 3' end.
- mRNA levels of replicative histones decrease at the end of S phase, and SLBP is degraded upon phosphorylation by cyclin A/CDK1
- The 3′ stem-loop structure is necessary and sufficient for the degradation of histone mRNAs
- Degradation is also regulated via 3′ end uridylation by the uridylyltransferase TUT7 and the 3′hExo enzyme is needed in the initial steps
- Human histone genes harbor TATA, CCAAT, and GC boxes in their promoter region, as well as putative binding sites for several transcription factors

# Distinct Transcriptional Regulation and Cell Cycle Expression of The Centromeric Variant CenH3

- **CenH3 CENP-A is encoded by a single multi-exon gene located outside of histone clusters.**
- **The expression of CenH3 CENP-A is regulated through a distinct, DNA synthesis-independent pathway.**
- Transcripts lack a 3' stem loop and undergo conventional processing through splicing and poly-adenylation.
- The expression of CenH3 CENP-A peaks in G2, and this temporal control is key for its centromeric targeting.
- A controlled expression of CenH3 CENP-A is necessary to prevent promiscuous interactions with low affinity chaperones and aberrant CenH3 CENP-A loading.
- The regulation of both CenH3 CENP-A and HJURP is coordinated in concert with other late cell cycle genes involved in mitotic progression, such as CDC25B, AURKB, PLK1, and CENP-B. Expression of these genes peak in G2/M.
- Exogenous over-expression of either CenH3 CENP-A or HJURP leads to an increase in the endogenous levels of their binding partner.
- CDK1/cyclin B–dependent phosphorylation at serine68 during G2/M hinders the interaction with HJURP, preventing premature loading of CenH3 CENP-A at centromeres

# Consitituve Expression of the Non-replicative H3.3 Variant by Independent Paralogs

- **H3.3 is redundantly encoded by two paralogous genes : A & B.**
- **These paralogs show different coding sequences, cis-regulatory targets and intron-exon organization.**
- **H3.3A & H3.3B are solitary genes lacking a 3' stem-loop structure. Both give rise to poly-adenylated transcripts, and H3.3B can generate up to three isoforms from alternative polyadenylation sites.**
- Although both H3.3A and H3.3B can contribute to new H3.3 synthesis, the two genes show distinct expression profiles in the male and female germ-line, among different tissues, and during development.
- Genes in the HIST1 cluster are significantly up-regulated across different breast cancer cell lines and breast tumor specimens
- **Differences in expression may affect the tissue specificity and outcome of specific amino acid substitutions**