# The Roles of Histone Variants in Fine-tuning Chromatin Organization and Function

Citation: Martire, S., & Banaszynski, L. A. (2020). The roles of histone variants in fine-tuning chromatin organization and function. Nature reviews. Molecular cell biology, 21(9), 522–541. https://doi.org/10.1038/s41580-020-0262-8

# Introduction

- Histone variants also have highly similar isoforms called ‘subvariants’ (encoded mainly by pseudogenes), which are not conserved among species and might play a role in regulating tissue-specific gene expression
- Besides the widely appreciated mechanisms such as histone and DNA modification, chromatin remodelling and topological higher-order chromatin organization, the incorporation of histone variants into chromatin contributes to the diversity of nucleosome structure and function

# H3 Variants

- In mammals, H3.3 is encoded by two distinct genes, H3F3A and H3F3B (also known as H3-3A and H3-3B), whose translation results in an identical protein product that differs from H3.1 and H3.2 by only four or five amino acids
- Histone chaperones are able to distinguish between H3.1/H3.2 and H3.3 variants using an amino acid motif at the base of α-helix 2 in the core of the protein, wherein the Ser-Ala-Val-Met motif from amino acids 87–90 in H3.1 and H3.2 is replaced by the Ala-Ala-Ile-Gly motif in H3.3.
- Another difference between H3.1/H3.2 and H3.3 occurs in the highly post-translationally modified N-terminal tail with Ala at position 31 in H3.1 and H3.2 that is replaced in H3.3 by a highly conserved Ser.
- H3.3 deposition is generally an indication of dynamic chromatin, with enriched regions exhibiting high turnover rates on a genome-wide level.
- Accordingly, H3.3 deposition is characteristic of euchromatin and specifically associated with active regulatory elements, including enhancers and promoters as well as gene bodies.
- In addition to H3.3, a number of additional replication-independent H3 variants exist. These variants may be expressed only in specific tissues, for example the testis-specific H3.4 (also known as H3T) and H3.5, and may be species specific, such as the primate-specific H3.Y.1 and H3.Y.2 (previously referred to as H3.Y and H3.X, respectively) and H3.5, which is specific to the hominid lineage.
- Despite containing the Ala-Ala-Ile-Gly motif, H3.Y.1 interacts only with the HIRA complex and not with ATRX–DAXX, which was found to be dependent on a Val46-to-Leu46 substitution from H3.3 to H3.Y.1
- CENP-A exhibits 50–60% sequence similarity with most other H3 variants in the histone-fold domain and possesses a unique N-terminal tail.

# H2A Variants

- To date, the H2A family contains the greatest sequence diversity of identified variants. In humans, four classes of replication-independent H2A variants have been reported: H2A.X, H2A.Z, macroH2A and testis-specific short H2A variants (H2A.B, H2A.L, H2A.P and H2A.Q)

# H2B and H4 Variants

- H4 and H2B contact only other histones
- Recently a new H4 variant, H4G, was found in a variety of human cell lines with localization primarily to the nucleoli
- Minor sequence differences are observed in the four H2B variants identified so far (H2B, H2BE, TSH2B and H2B.W), which typically involve only a few amino acid substitutions

# Nucleosome Structure and Dynamics

- One aspect of incorporation of histone variants into nucleosomes is changes to their structure and stability.
- Functionally, H2A.B deposition is correlated with active transcription and splicing both in vitro and in vivo
- H3.3 is highly enriched in euchromatin, which is characterized by high levels of histone acetylation and is strongly correlated with active transcription.
- Loss of H3.3 results in reduced recruitment of the nucleosome remodelling deacetylase (NuRD) chromatin remodelling complex to active promoters

# Variant-specific Chromatin States

- Generally, the PTMs that histone variants carry can be considered a product of their localization to specific regions, and it can be difficult to decouple this effect from functions that can be directly attributed to the histone variant itself.
- Proteomics studies have shown that H3.3 is enriched with modifications associated with genome activation, such as H3K4me3, H3K36me3 and H3K27ac.
- A histone variant-specific PTM is the phosphorylation of a unique Ser31 on the H3.3 N-terminal tail. This phosphorylation was originally reported as a mitotic mark , but several studies suggest that this serine and/or its phosphorylation are important developmentally and are linked to gene regulation.
- Variant-specific chaperones can influence the composition of the nucleo­some; for example, ATRX prevents the deposition of macroH2A in a mechanism that is independent of ATRX association with DAXX

# Relevance to Human Disease

- Transcriptional repression of macroH2A in melanoma contributes to upregulation of oncoproteins that drive cell proliferation, such as cyclin-dependent kinase 8 (CDK8), and results in altered metabolism